src/Cli/dotnet/CommonOptions.cs

```diff
@@ -174,6 +175,44 @@ internal static string ArchOptionValue(ParseResult parseResult) =>
             // Flip the argument so that if this option is specified we get selfcontained=false
             .SetForwardingFunction((arg, p) => ForwardSelfContainedOptions(!arg, p));
 
+        public static readonly CliOption<IReadOnlyDictionary<string, string>> EnvOption = new("--environment", "-e")
+        {
+            Description = CommonLocalizableStrings.CmdEnvironmentVariableDescription,
+            HelpName = CommonLocalizableStrings.CmdEnvironmentVariableExpression,
+            CustomParser = ParseEnvironmentVariables,
+            // Can't allow multiple arguments because the separator needs to be parsed as part of the environment variable value.
+            AllowMultipleArgumentsPerToken = false
+        };
+
+        private static IReadOnlyDictionary<string, string> ParseEnvironmentVariables(ArgumentResult argumentResult)
+        {
+            Dictionary<string, string> result = [];
+            List<CliToken>? invalid = null;
+
+            foreach (var token in argumentResult.Tokens)
+            {
+                if (token.Value.IndexOf('=') is var index and > 0 &&
+                    token.Value[0..index].Trim() is var name and not "")
+                {
+                    result[name] = token.Value[(index + 1)..];
```

Comment:
On Windows, environment variable names are typically case insensitive, so if a user passes `x=1` and `X=2` we'd have two dictionary entries. There'd only be one variable in the PEB, but then the final value depends on the order it was passed to the process. On non-Windows environments, the names are case sensitive, so I'm guessing we trust the user's intent when there are multiple pairs?